api/v1beta1/azurecluster_validation.go

```diff
@@ -136,12 +141,24 @@ func validateBastionSpec(bastionSpec BastionSpec, fldPath *field.Path) *field.Er
 	return nil
 }
 
+// validateIdentityRef validates an IdentityRef.
+func validateIdentityRef(identityRef *corev1.ObjectReference, fldPath *field.Path) *field.Error {
+	if identityRef == nil {
+		return field.NotFound(fldPath, "identityRef is required")
+	}
+	if identityRef.Kind != "AzureClusterIdentity" {
+		return field.Invalid(fldPath.Child("name"), identityRef.Name,
```

Comment:
nit: `field.NotSupported` is slightly more appropriate IMO

> // NotSupported returns a *Error indicating "unsupported value".
// This is used to report unknown values for enumerated fields (e.g. a list of
// valid values).

> // Invalid returns a *Error indicating "invalid value".  This is used
// to report malformed values (e.g. failed regex match, too long, out of bounds).